src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp

```diff
@@ -708,24 +708,24 @@ void OpenGLDrawingContext::DrawSprite(uint32_t image, int32_t x, int32_t y, uint
     bool special = false;
     if (image & IMAGE_TYPE_REMAP_2_PLUS)
     {
-        palettes.x = TextureCache::PaletteToY((image >> 19) & 0x1F);
-        palettes.y = TextureCache::PaletteToY((image >> 24) & 0x1F);
+        palettes.x = TextureCache::PaletteToY(static_cast<FilterPaletteID>((image >> 19) & 0x1F));
```

Comment:
If it is anded with 0x1F it is a Colour ID. PaletteID's are 0xFF. The problem though is that all colours are Palettes but not all Palettes are colours. I'm still not sure on this at all what the best approach is.

Reply:
PalleteToY returns GLint. According to my IDE this was used only with FilterPaletteID enum and thus the function signature was updated to take this enum. There is an option to revert it to earlier state and always cast strong enums to functions type.